### PR TITLE
3.0 - cleanup

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: src/Fixer/Basic/NonPrintableCharacterFixer.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ protected function createConfigurationDefinition()
         return new FixerConfigurationResolver([
             (new FixerOptionBuilder('use_escape_sequences_in_strings', 'Whether characters should be replaced with escape sequences in strings.'))
                 ->setAllowedTypes(['bool'])
-                ->setDefault(false) // @TODO 3.0 consider changing to true, but it will require that all fixed code by default is php7+, maybe the best is to keep it in `@PHP70Migration:risky` only
+                ->setDefault(\PHP_VERSION_ID >= 70000)
```
Comment (keradus) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php:107]:
```suggestion
                ->setDefault(true)
```

you may explicitly set `true` here.
v3 won't suppor PHP 5.x

File: tests/Fixer/ConstantNotation/NativeConstantInvocationFixerTest.php [hunk 2]
```diff
@@ -217,7 +217,7 @@ public function provideFix71WithDefaultConfigurationCases()
             [
                 '<?php
 try {
-    foo(\JSON_ERROR_DEPTH|\JSON_PRETTY_PRINT|\JOB_QUEUE_PRIORITY_HIGH);
+    foo(\JSON_ERROR_DEPTH|\JSON_PRETTY_PRINT|JOB_QUEUE_PRIORITY_HIGH);
```
Comment (keradus) [inline tests/Fixer/ConstantNotation/NativeConstantInvocationFixerTest.php:220]: why the change?

File: UPGRADE-v3.md [hunk 3]
```diff
@@ -145,4 +148,5 @@ Code BC changes
 - class `Tokens` is now `final`
 - method `create` of class `Config` has been removed, use the constructor
 - method `create` of class `RuleSet` has been removed, use the constructor
+- method `getClassyElements` of class `TokensAnalyzer` parameter `$returnTraitsImports` has been removed; now always returns trait import information
```
Comment (keradus) [inline UPGRADE-v3.md:151]: it's internal class, so it's not BC break

File: tests/Fixer/ConstantNotation/NativeConstantInvocationFixerTest.php (hunk 2 above)
Comment (keradus) [reply tests/Fixer/ConstantNotation/NativeConstantInvocationFixerTest.php:220]: i see now

File: src/Fixer/Basic/NonPrintableCharacterFixer.php (hunk 1 above)
Comment (keradus) [reply src/Fixer/Basic/NonPrintableCharacterFixer.php:107]: so at least
```suggestion
                ->setDefault(\PHP_VERSION_ID >= 70000) // @TODO 3.0 replace with `true` when changing PHP requirement to 7+
```

File: UPGRADE-v3.md (hunk 3 above)
Comment (keradus) [reply UPGRADE-v3.md:151]: at least this, maybe
let's take this and merge the PR, but I'm happy for further discussion (and maybe even drop the notes about internal classes changes)

```suggestion

### BC breaks; common internal classes

- method `getClassyElements` of class `TokensAnalyzer` parameter `$returnTraitsImports` has been removed; now always returns trait import information
```

**Review by keradus — APPROVED** [review]
nice one!

**Comment by keradus** [comment]
Thank you @SpacePossum.

File: UPGRADE-v3.md (hunk 3 above)
Comment (SpacePossum) [reply UPGRADE-v3.md:154]: Maybe when v3 is real close for release we can broom this file again and remove overcomplete details :+1:
